doc/decisions/man_pages.md

```diff
@@ -0,0 +1,76 @@
+# Man Pages
+
+## Problem
+
+Our man pages are written as Markdown in doc/help and then converted to roff and stored in doc/man.
+This was a workaround, because `ronn-ng` isn't available on most distributions and we want to avoid packages without man pages are being built.
+Storing generated files is annoying, as it requires:
+
+- developers to always update generated files if the sources are changed
+- developers not committing irrelevant changes to generated files (e.g. as may occur with different CMAKE_INSTALL_PREFIX etc.)
+- require extra effort for continuous integration, e.g. [#4542](https://issues.libelektra.org/4542)
+
+## Constraints
+
+1. we want beautiful rendered man pages, e.g., OPTIONS section looks like normal man pages, see in Notes¹ below
+2. we cannot require rare tools for the build process: the man pages must be present in every package
+
+## Assumptions
+
+## Considered Alternatives
+
+0. `ronn-ng`, which doesn't have packages on most distribution (violates constraint 2.) and thus created this problem
```

Comment:
At least it can be installed with `gem` (see https://github.com/apjanke/ronn-ng/blob/master/INSTALLING.md) which should be available on most distributions.
So I'm not sure if the availability of the tool should really be considered that bad.

Reply:
Many distributions require that they can build packages exclusively with packages they have. Is this different in Gentoo?

Reply:
I think this just needs clarification that using `ronn-ng` **and** not committing the generated files is not possible because of this constraint. The current state on `master` is not a problem, unless we want to remove the generated files.

Reply:
Hm, looks like I'm propably missing something here. :confused:

Packages that are offered by the package-managers of distributions usually must be created by a package maintainer.
Most distributions use pre-compiled binaries, here Gentoo is an exception where almost everything gets compiled locally.

So if package maintainers have to compile the software and package the binaries, is it a problem if they also must build the manpages in the process?
For developers and maintainers I think it should be ok if they have to install a software like ronn-ng manually.
Is this the point we are actually talking about here?
I've also read the discussion in https://github.com/ElektraInitiative/libelektra/issues/4542

End-users then get the pre-compiled packages with binaries and manpages from their distribution.

So the only real drawback I can see here if is someone wants
to build the package locally as mentioned in https://www.libelektra.org/installation/installation#os-independent.
But as compilation steps and installation of other utilities like debhelper is also necessary during
this process, a additional `gem install` doesn't look like a big problem to me.

But as I said, I'm propably missing something here, so I'd welcome clarification, thanks!

Reply:
Thx @kodebach the two interwoven problems were not explained well, I changed in f86628ce541a2a451eca393a758d083a643e0a59

@flo91 I gave an example about deb/dpkg but it would surprise me if it would be completely different in Gentoo. When you type `emerge elektra`, it might emerge other packages but it would be at least very bad style if `emerge elektra` itself would do a `gem install ronn-ng`. Would it be allowed that a `emerge ronn-ng` simply does a `gem install ronn-ng`? If this would be allowed, I have to admit that in Gentoo the `ronn-ng` problem is not a very though one. For most distributions, however, where alternative package installation methods are not allowed, packaging a ruby application is non-trivial. Hence most distribution don't have `ronn-ng` packaged. Debian actually has a [`ronn` packaged but it is an older version](https://packages.debian.org/sid/ronn). I clarified this in b5baca12a0ca875d518cd1d684654e92049d5ac8.

Reply:
First of all, I never was a package-maintainer and never added a package to a distribution, so I'm definitely
not an expert in this topic, but I want to explain my understanding.

`ronn-ng` is basically a **build-time** dependency, not a run-time dependency.
So the package maintainers (who install `ronn-ng` alongside with the
required compilers, linkers, libs, etc.) could create the man pages and package them alongside with the compiled
binaries. The end-users who install the package, e.g. via `apt-get install libelektra`, don't need `ronn-ng` on
their local system because the package-maintainers already built the man pages.


So basically it would move the step of using `ronn-ng` to create the man pages to the package maintainer.
AFAIK in Gentoo you can also only define dependencies that are available via portage, but I think in this
case it should be also possible to package the already created man pages.
However, I'm focusing more on binary-based distributions like Debian here.



Your wrote in the decision text: 

> "But we want to avoid that distributions build packages without man pages, hence we added the generated files."

OK, so you don't want that the package **maintainers** have to install `ronn-ng` on their systems.
If that's the case, another possibility could be to create the man pages and add them to our repository for every release, but not in the `master` branch, so the building of man pages would be a part of the release process.
Package maintainers can then checkout the releases and get the appropriate man pages.

Reply:
I'm also no expert, but I think markus2330 wanted to say that on Debian it must be possible to build the package on Debian without only packages available on Debian. At least, if we want the package to be in the Debian stable repo. Of course we can still build our own `.deb` packages as we do now.

> building of man pages would be a part of the release process

Only having the man pages as part of the release artifacts (maybe both as a separate `manpages.zip` and in the main release artifact) would be better than want we have now. But I don't know if it fulfils the Debian criteria. Maybe they just want to checkout a git commit and build from that. In that case the releases are not a solution.

Reply:
Exactly, like @kodebach says it does not really matter which kind of dependency it is. For integration into the archive all build and run-time dependencies must be present as packages. For Debian, and probably most other distributions, there must be a reproducible way how to compile a package (e.g. dpkg-buildpackage). And as part of that reproducibility, only packages can be allowed as any dependency.

As this is basics for maintainers and not really relevant for developers, I will not add it to the decision.

> If that's the case, another possibility could be to create the man pages and add them to our repository for every release, but not in the master branch, so the building of man pages would be a part of the release process.
Package maintainers can then checkout the releases and get the appropriate man pages.

Yes, this would be a different solution, with other drawbacks we can discuss when someone creates a decision PR for that.

Reply:
> > If that's the case, another possibility could be to create the man pages and add them to our repository for every release, but not in the master branch, so the building of man pages would be a part of the release process.
> > Package maintainers can then checkout the releases and get the appropriate man pages.
> 
> Yes, this would be a different solution, with other drawbacks we can discuss when someone creates a decision PR for that.

Wouldn't this just be a "Considered Alternative" for this decision? It would solve the problem described here AFAICT.

Reply:
Ok, thank you both for the helpful clarifications! :smiley:
After thinking about it, I understand that reproducibility of the build process with using only tools provided by the distribution can help the maintainers ensuring stability and maintainability.


> Wouldn't this just be a "Considered Alternative" for this decision? It would solve the problem described here AFAICT.

For me, it also looks more like a "Considered Alternative" than a whole new decision.

Reply:
As the TODO said I actually only wanted to clarify the problem, not the potential solutions, which should be done later by @hannes99 in a real decision PR. The update of the manpage decision in this PR was a fault on my side, it clearly showed to me that decision PRs should only contain one decision. :wink: 

I did an update nevertheless: 99cb07833852b9955e4690566e96f7ed404b4dab